### ISSUE TITLE
v4 VS v8  

### ISSUE THREAD

**Issue by hardik2396** [shubhamagarwal92/mmd#6]
Can you tell me what is a performance difference between config version 4 and config version 8 ? 

Thanks in advance

**Comment by githwd2016** [comment]
When using config version 4, I got the similar result, i.e., BLEU-4=0.4321. However, when config version 8 was used, I got BLEU-4=0!!! The model just outputs <EOS>. It seems that the biggest difference between two versions is whether using attention. Do you got the similar result with me?

**Comment by shubhamagarwal92** [comment]
Though attention improves our results (in contrast to Shah et al.), BLEU should not be 0. I can cross check. Are you sure, there is no error on your part?

**Comment by githwd2016** [comment]
I just use the code provided by you and it works when using config version 4. I check the difference between two versions. Besides attention, the **keep probability** is also different(version 4 is 0.3 and version 8 is 0.8). Does this matter?
I also check the parameters of decoder, and find something strange(epoch 10).
<img width="631" alt="屏幕快照 2019-06-06 上午9 46 36" src="https://user-images.githubusercontent.com/20674430/59001325-b22ba500-8840-11e9-98da-b5b84d8bd422.png">

**Comment by githwd2016** [comment]
Moreover, the loss of version 8 does not decrease. It keeps at level 0.01.
<img width="750" alt="屏幕快照 2019-06-06 上午9 59 00" src="https://user-images.githubusercontent.com/20674430/59001598-c0c68c00-8841-11e9-8a1d-869eeaa0c68a.png">
However, in version 4, the loss decreases obviously.
<img width="744" alt="屏幕快照 2019-06-06 上午9 57 09" src="https://user-images.githubusercontent.com/20674430/59001614-d936a680-8841-11e9-8c1a-c2f9a1e9b568.png">

**Comment by shubhamagarwal92** [comment]
@githwd2016 I added another config with the logs and metrics. https://github.com/shubhamagarwal92/mmd/commit/af1550eec2eb9c6cc74838cbc851077d3bffb9b3 Please try and see if this works.

Try the config with
```
Multimodal HRED 
context 5
epoch 8 for evaluation
```
I hope this helps.

**Comment by githwd2016** [comment]
Thanks. I have got the correct result. However, there is still a problem.
In **translate.py**, when I use **getattr** to choose model, the result only contain two tokens, like this:
<img width="117" alt="屏幕快照 2019-06-13 下午1 44 52" src="https://user-images.githubusercontent.com/20674430/59406627-71ec9980-8de1-11e9-9154-863c3047daf5.png">
But when I use **if...else...**, it become normal.
<img width="431" alt="屏幕快照 2019-06-13 下午1 48 25" src="https://user-images.githubusercontent.com/20674430/59406815-f808e000-8de1-11e9-8219-1d26a7b9d80d.png">

**Comment by shubhamagarwal92** [comment]
This is really strange. Are you sure this was the only change? `getattr` should ideally do the same thing. Probably related to [max_decode_len](https://github.com/shubhamagarwal92/mmd/blob/master/config/config_hred_mmd_v3.json#L43) flag?

**Comment by githwd2016** [comment]
I didn't change the config file. And I even clone the code again and only change this.
Anyway, I got the result. Thanks for your patience.

**Comment by shubhamagarwal92** [comment]
> When using config version 4, I got the similar result, i.e., BLEU-4=0.4321. However, when config version 8 was used, I got BLEU-4=0!!! The model just outputs . It seems that the biggest difference between two versions is whether using attention. Do you got the similar result with me?

Also if you compare both the configs, you may see that the difference was also of `training` hyper-parameters and `dropouts` (apart from using attention). I somehow put the wrong config earlier (but actually wanted to show the difference with attention). Sorry for that! But it is still strange that you were getting Bleu 0.

**Comment by shubhamagarwal92** [comment]
> I didn't change the config file. And I even clone the code again and only change this.
> Anyway, I got the result. Thanks for your patience.

Could you open this PR as well (just comment out the `getattr`; I will debug it later if I get some time). Thanks again :)

**Comment by githwd2016** [comment]
I have opened a PR.

**Comment by shubhamagarwal92** [comment]
Thanks for the PR. Closing this now. Let me know (or re-open this) if you face any other issue.
